src/Integration.Vsix/SonarLintTracker/IssuesSnapshot.cs

```diff
@@ -0,0 +1,124 @@
+using System.Collections.ObjectModel;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    internal class IssuesSnapshot : WpfTableEntriesSnapshotBase
+    {
+        private readonly string filePath;
+        private readonly int versionNumber;
+
+        private readonly IList<IssueMarker> issueMarkers;
+        private readonly IReadOnlyCollection<IssueMarker> readonlyIssueMarkers;
+        public IEnumerable<IssueMarker> IssueMarkers => readonlyIssueMarkers;
+
+        public IssuesSnapshot NextSnapshot;
+
+        internal IssuesSnapshot(string filePath, int versionNumber, IEnumerable<IssueMarker> issueMarkers)
+        {
+            this.filePath = filePath;
+            this.versionNumber = versionNumber;
+            this.issueMarkers = new List<IssueMarker>(issueMarkers);
+            this.readonlyIssueMarkers = new ReadOnlyCollection<IssueMarker>(this.issueMarkers);
+        }
+
+        public override int Count => this.issueMarkers.Count;
+
+        public override int VersionNumber => this.versionNumber;
+
+        public override bool TryGetValue(int index, string columnName, out object content)
+        {
+            if (index < 0 || this.issueMarkers.Count <= index)
+            {
+                content = null;
+                return false;
+            }
+
+            switch (columnName)
+            {
+                case StandardTableKeyNames.DocumentName:
+                    content = filePath;
+                    return true;
+
+                case StandardTableKeyNames.Line:
+                    content = this.issueMarkers[index].Span.Start.GetContainingLine().LineNumber;
+                    return true;
+
+                case StandardTableKeyNames.Column:
+                    var position = this.issueMarkers[index].Span.Start;
+                    var line = position.GetContainingLine();
+                    content = position.Position - line.Start.Position;
+                    return true;
+
+                case StandardTableKeyNames.Text:
+                    content = this.issueMarkers[index].Issue.Message;
+                    return true;
+
+                case StandardTableKeyNames.ErrorSeverity:
+                    // TODO use correct icon type depending on severity
+                    content = __VSERRORCATEGORY.EC_WARNING;
+                    return true;
+
+                case StandardTableKeyNames.BuildTool:
+                    // TODO get correct analyzer name
+                    content = "SonarJS [SonarLint for Visual Studio 2015]";
+                    return true;
+
+                case StandardTableKeyNames.ErrorCode:
+                    content = this.issueMarkers[index].Issue.RuleKey;
+                    return true;
+
+                case StandardTableKeyNames.ErrorCodeToolTip:
+                case StandardTableKeyNames.HelpLink:
+                    // TODO use correct version
+                    // TODO add JavaScript rules on website
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)